packages/flutter/lib/src/widgets/adapter.dart

```diff
@@ -0,0 +1,179 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'framework.dart';
+
```

Comment:
I should point out that the code in this file was copied one-to-one from widgets/binding.dart. It implements the old way of bootstrapping the widget tree by attaching it to a RenderView. It still works and the widget has some very limited use in the wild, so I decided to keep it around.

I did modify some of the doc comments to explain that this isn't used by the framework itself anymore and how it relates to the new kid in town "RootWidget" that took its job.

I also changed the base class of the element to RenderTreeRootElement because this is functioning as the root of a render tree as well.

Reply:
Shall we make extracting this class a separate PR to make this PR cleaner?

Reply:
I originally considered it, but it actually makes the diff for the RootWidget (newly added in this PR) harder. Now, its pretty obvious to see how the RootWidget evolved from the Adapter. If we move the adapter first, the RootWidget code will show up all as newly added lines...